### DISCUSSION TITLE
Workflow compare with google/release-please-action?

### DISCUSSION THREAD

**Discussion by linrongbin16** [orhun/git-cliff#593] [Q&A] [answered]
hi, would you please compare git-cliff with [release-please-action](https://github.com/google-github-actions/release-please-action). now I am using it with github actions, to automatically release project.

(in case we are not on the same page, let me introduce it a little)

it will run on every push for `main` branch, collect the conventional commits e.g. `feat`/`fix`/`perf` etc, and submit a release PR. The PR contains a changelog changes, which is highly similar to what git-cliff does (but seems not as extendable as git-cliff, I actually prefer the commit author and PR number examples in git-cliff).

I can either merge it or leave it there. if I leave it there, in following push on `main` branch, release-please-action will use force push to update the PR (with newer commits).

if I merge the release PR to main branch, it will trigger the release-please-action to do another thing: it will create a git tag (e.g. next version, for example `v1.3.2`), and publish the release.

since I am trying to create a rust/cargo project, and I can see that cargo community has its own popular release tools: git-cliff for generate the changelog, and [cargo-release](https://github.com/crate-ci/cargo-release) for publish the release and upload the cargo to crates.io.

but how should I integrate these two tools with github actions? especially for the 2-steps: submit release PR and publish after merge release PR? cargo-release docs actually didn't say too much about it, I am confused if I could do the same workflow with git-cliff+cargo-release.

**Comment by orhun** [comment, marked as answer]
Hey!

Check out [release-plz](https://github.com/MarcoIeni/release-plz) by @MarcoIeni, it does exactly what you described. It also supports `git-cliff` (although we still need to work on the GitHub integration 🙂 )

**Comment by powerman** [comment]
Is there anything like [release-plz](https://github.com/MarcoIeni/release-plz) by @MarcoIeni but not tied to Rust projects? I.e. just create/update "release please" PR and handle merging that PR by just creating tag and GitHub release with git-cliff's changelog - and nothing more! Other project/language-specific jobs/workflows will be triggered by created release and do everything else (build binaries, upload artifacts to GitHub release and/or language/OS-specific package managers etc.).

Only implementation I was able to find so far is a manual script like [this one](https://github.com/jdx/mise/blob/main/tasks/release-plz).

**Reply by marcoieni** [reply]
you don't like https://github.com/googleapis/release-please-action ?

**Reply by powerman** [reply]
Honestly - it just feels wrong to me. It does too much, it has too many open bug-issues, it feels too complicated. I suppose it will works just fine in most cases, but when it wont - I expect this to became a real issue.

**Reply by marcoieni** [reply]
I see. Good luck with your research then :)

**Reply by powerman** [reply]
Well, I implemented my own "Release PR" solution as a reusable workflow: https://github.com/powerman/workflows
